src/Factory/ImageFactory.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace App\Factory;
+
+use App\Entity\Image as ImageEntity;
+use App\Repository\ImageRepository;
+
+final class ImageFactory
+{
+    public function __construct(
+        private readonly ImageRepository $imageRepository,
+    ) {
+    }
+
+    public function createOrUpdate(string $url, ?ImageEntity $image): ImageEntity
+    {
+        if (is_null($image)) {
+            $image = new ImageEntity();
+            $image->setSource($url);
+        } elseif ($image->getSource() !== $url) {
+            $image = new ImageEntity();
+            $image->setSource($url);
```

Comment:
```suggestion
        if (is_null($image) || $image->getSource() !== $url) {
            $image = new ImageEntity();
            $image->setSource($url);
```

or the hardcore version:

```suggestion
        if ($image?->getSource() !== $url) {
            $image = new ImageEntity();
            $image->setSource($url);
```

which works because `$url` is not `null`.